app/dashboard/src/components/Stepper/useStepperState.ts

```diff
@@ -70,6 +70,9 @@ export function useStepperState(props: StepperStateProps): UseStepperStateResult
     direction: Direction
   }>(() => ({ current: defaultStep, direction: 'initial' }))
 
+  const onStepChangeEventCallback = eventCallbackHooks.useEventCallback(onStepChange)
```

Comment:
i would prefer calling this `onStepChange` and the outer one either ` onStepChangeRaw` or using `props.onStepChange` directly.
mostly personal preference but i think the `EventCallback` suffix doesn't really make the variable name more descriptive and makes it a *lot* longer (and a bit harder to read)